packages/medusa/src/models/note.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  Entity,
+  BeforeInsert,
+  Column,
+  DeleteDateColumn,
+  CreateDateColumn,
+  UpdateDateColumn,
+  Index,
+  JoinColumn,
+  PrimaryColumn,
+  ManyToOne,
+} from "typeorm"
+import { ulid } from "ulid"
+import { User } from "./user"
+import { resolveDbType, DbAwareColumn } from "../utils/db-aware-column"
+
+@Entity()
+export class Note {
+  @PrimaryColumn()
+  id: string
+
+  @Column()
+  value: string
+
+  @Index()
+  @Column()
+  resource_type: string
+
+  @Index()
+  @Column()
+  resource_id: string
+
+  author_id: string
```

Comment:
Should be `@Column({ nullable: true })` 

Why nullable? 
In case we need to make it possible for plugins/3rd parties to add notes we should allow notes without a user